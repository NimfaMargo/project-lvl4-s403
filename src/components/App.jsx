import React from 'react';
import Sidebar from './Sidebar.jsx';
import MessagesList from './MessagesList.jsx';
import AddMessage from './AddMessage.jsx';

const App = () => (
  <div>
    <div className="row justify-content-around" style={{ height: '550px' }}>
      <div className="sidebar col-sm-12 col-md-3 col-xl-3" style={{ border: '3px solid #bddfd3' }}>
        <Sidebar/>
      </div>

      <div className="main d-flex flex-column col-sm-12 col-md-9 col-xl-9 mh-100" style={{ border: '3px solid #bddfd3' }}>
        <div className="col message-list overflow-auto bg-light align-self-stretch border m-2">
          <MessagesList />
        </div>
      </div>
    </div>
    <div className="row second-row" style={{ border: '3px solid #bddfd3' }}>
      <div className="col-sm-12 col-md-9 offset-md-3 input">
        <AddMessage />
      </div>
    </div>
  </div>

);
export default App;
