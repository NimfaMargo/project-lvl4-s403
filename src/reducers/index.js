import { combineReducers } from 'redux';
import { handleActions } from 'redux-actions';
import { reducer as formReducer } from 'redux-form';
import * as actions from '../actions';

// const messageAddingState = handleActions({
//   [actions.addMessageRequest]() {
//     return 'requested';
//   },
//   [actions.addMessageFailure]() {
//     return 'failed';
//   },
//   [actions.addMessageSuccess]() {
//     return 'finished';
//   },
// }, 'none');

const channels = handleActions({
  [actions.addChannel](state, channel) {
    return [...state, channel.payload];
  },
}, []);

const currentChannelId = handleActions({
}, []);

const messages = handleActions({
  [actions.addMessage](state, message) {
    return [...state, message.payload];
  },
}, []);

export default combineReducers({
  channels,
  messages,
  currentChannelId,
  form: formReducer,
});
