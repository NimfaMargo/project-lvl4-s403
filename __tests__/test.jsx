import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducers from '../src/reducers';
import App from '../src/components/App.jsx';

Enzyme.configure({ adapter: new Adapter() });

test('Store', () => {
  const store = createStore(reducers);

  const vdom = (
    <Provider store={store}>
      <App />
    </Provider>
  );
  const wrapper = mount(vdom);
  expect(wrapper.render()).toMatchSnapshot();
});
