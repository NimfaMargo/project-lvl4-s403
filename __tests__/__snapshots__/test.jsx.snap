// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store 1`] = `
<div
  class="container"
>
  <div
    class="row justify-content-around h-100"
  >
    <div
      class="col-sm-12 col-md-4 col-xl-4 sidebar"
    >
      <div
        class="container"
      >
        <div
          class="sidebar-header mt-3"
        >
          <h4
            class="text-center text-white"
          >
            Channels
          </h4>
        </div>
        <form>
          <div
            class="input-group m-2"
          >
            <input
              class="form-control"
              maxlength="17"
              name="text"
              required=""
              type="channel"
              value=""
            />
            <div
              class="input-group-append"
            >
              <button
                class="btn btn-info btn-sm channel"
                type="submit"
              >
                Add
              </button>
            </div>
          </div>
        </form>
        <div
          class="channels-list p-0 m-0 overflow-auto"
          style="height: 500px;"
        >
          <ul
            class="nav nav-pills flex-column ml-4"
          />
        </div>
      </div>
    </div>
    <div
      class="col-sm-12 col-md-8 col-xl-8 main p-0"
    >
      <div
        class="container h-100"
      >
        <div
          class="row position-absolute w-100"
          style="z-index: 999; display: block;"
        />
        <div
          class="row message-list overflow-auto bg-light"
          style="height: 550px;"
        >
          <div
            class="messages w-100 pl-4 pr-4"
          >
            <div />
          </div>
        </div>
        <div
          class="row bg-light"
          style="height: 55px;"
        >
          <form
            class="w-100 pl-2 pr-2"
          >
            <div
              class="input-group mt-2"
            >
              <input
                class="form-control"
                name="text"
                required=""
                type="text"
                value=""
              />
              <div
                class="input-group-append"
              >
                <button
                  class="btn btn-info btn-sm message"
                  type="submit"
                >
                  Send
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Store 2`] = `
<div
  class="container"
>
  <div
    class="row justify-content-around h-100"
  >
    <div
      class="col-sm-12 col-md-4 col-xl-4 sidebar"
    >
      <div
        class="container"
      >
        <div
          class="sidebar-header mt-3"
        >
          <h4
            class="text-center text-white"
          >
            Channels
          </h4>
        </div>
        <form>
          <div
            class="input-group m-2"
          >
            <input
              class="form-control"
              maxlength="17"
              name="text"
              required=""
              type="channel"
              value=""
            />
            <div
              class="input-group-append"
            >
              <button
                class="btn btn-info btn-sm channel"
                type="submit"
              >
                Add
              </button>
            </div>
          </div>
        </form>
        <div
          class="channels-list p-0 m-0 overflow-auto"
          style="height: 500px;"
        >
          <ul
            class="nav nav-pills flex-column ml-4"
          />
        </div>
      </div>
    </div>
    <div
      class="col-sm-12 col-md-8 col-xl-8 main p-0"
    >
      <div
        class="container h-100"
      >
        <div
          class="row position-absolute w-100"
          style="z-index: 999; display: block;"
        />
        <div
          class="row message-list overflow-auto bg-light"
          style="height: 550px;"
        >
          <div
            class="messages w-100 pl-4 pr-4"
          >
            <div />
          </div>
        </div>
        <div
          class="row bg-light"
          style="height: 55px;"
        >
          <form
            class="w-100 pl-2 pr-2"
          >
            <div
              class="input-group mt-2"
            >
              <input
                class="form-control"
                name="text"
                required=""
                type="text"
                value="na-na"
              />
              <div
                class="input-group-append"
              >
                <button
                  class="btn btn-info btn-sm message"
                  type="submit"
                >
                  Send
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Store 3`] = `
<div
  class="container"
>
  <div
    class="row justify-content-around h-100"
  >
    <div
      class="col-sm-12 col-md-4 col-xl-4 sidebar"
    >
      <div
        class="container"
      >
        <div
          class="sidebar-header mt-3"
        >
          <h4
            class="text-center text-white"
          >
            Channels
          </h4>
        </div>
        <form>
          <div
            class="input-group m-2"
          >
            <input
              class="form-control"
              maxlength="17"
              name="text"
              required=""
              type="channel"
              value=""
            />
            <div
              class="input-group-append"
            >
              <button
                class="btn btn-info btn-sm channel"
                type="submit"
              >
                Add
              </button>
            </div>
          </div>
        </form>
        <div
          class="channels-list p-0 m-0 overflow-auto"
          style="height: 500px;"
        >
          <ul
            class="nav nav-pills flex-column ml-4"
          />
        </div>
      </div>
    </div>
    <div
      class="col-sm-12 col-md-8 col-xl-8 main p-0"
    >
      <div
        class="container h-100"
      >
        <div
          class="row position-absolute w-100"
          style="z-index: 999; display: block;"
        />
        <div
          class="row message-list overflow-auto bg-light"
          style="height: 550px;"
        >
          <div
            class="messages w-100 pl-4 pr-4"
          >
            <div />
          </div>
        </div>
        <div
          class="row bg-light"
          style="height: 55px;"
        >
          <form
            class="w-100 pl-2 pr-2"
          >
            <div
              class="input-group mt-2"
            >
              <input
                class="form-control"
                disabled=""
                name="text"
                required=""
                type="text"
                value="na-na"
              />
              <div
                class="input-group-append"
              >
                <button
                  class="btn btn-info btn-sm message"
                  disabled=""
                  type="submit"
                >
                  Send
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Store 4`] = `
<div
  class="container"
>
  <div
    class="row justify-content-around h-100"
  >
    <div
      class="col-sm-12 col-md-4 col-xl-4 sidebar"
    >
      <div
        class="container"
      >
        <div
          class="sidebar-header mt-3"
        >
          <h4
            class="text-center text-white"
          >
            Channels
          </h4>
        </div>
        <form>
          <div
            class="input-group m-2"
          >
            <input
              class="form-control"
              maxlength="17"
              name="text"
              required=""
              type="channel"
              value=""
            />
            <div
              class="input-group-append"
            >
              <button
                class="btn btn-info btn-sm channel"
                type="submit"
              >
                Add
              </button>
            </div>
          </div>
        </form>
        <div
          class="channels-list p-0 m-0 overflow-auto"
          style="height: 500px;"
        >
          <ul
            class="nav nav-pills flex-column ml-4"
          />
        </div>
      </div>
    </div>
    <div
      class="col-sm-12 col-md-8 col-xl-8 main p-0"
    >
      <div
        class="container h-100"
      >
        <div
          class="row position-absolute w-100"
          style="z-index: 999; display: block;"
        />
        <div
          class="row message-list overflow-auto bg-light"
          style="height: 550px;"
        >
          <div
            class="messages w-100 pl-4 pr-4"
          >
            <div />
          </div>
        </div>
        <div
          class="row bg-light"
          style="height: 55px;"
        >
          <form
            class="w-100 pl-2 pr-2"
          >
            <div
              class="input-group mt-2"
            >
              <input
                class="form-control"
                disabled=""
                name="text"
                required=""
                type="text"
                value="another task"
              />
              <div
                class="input-group-append"
              >
                <button
                  class="btn btn-info btn-sm message"
                  disabled=""
                  type="submit"
                >
                  Send
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Store 5`] = `
<div
  class="container"
>
  <div
    class="row justify-content-around h-100"
  >
    <div
      class="col-sm-12 col-md-4 col-xl-4 sidebar"
    >
      <div
        class="container"
      >
        <div
          class="sidebar-header mt-3"
        >
          <h4
            class="text-center text-white"
          >
            Channels
          </h4>
        </div>
        <form>
          <div
            class="input-group m-2"
          >
            <input
              class="form-control"
              maxlength="17"
              name="text"
              required=""
              type="channel"
              value=""
            />
            <div
              class="input-group-append"
            >
              <button
                class="btn btn-info btn-sm channel"
                type="submit"
              >
                Add
              </button>
            </div>
          </div>
        </form>
        <div
          class="channels-list p-0 m-0 overflow-auto"
          style="height: 500px;"
        >
          <ul
            class="nav nav-pills flex-column ml-4"
          />
        </div>
      </div>
    </div>
    <div
      class="col-sm-12 col-md-8 col-xl-8 main p-0"
    >
      <div
        class="container h-100"
      >
        <div
          class="row position-absolute w-100"
          style="z-index: 999; display: block;"
        />
        <div
          class="row message-list overflow-auto bg-light"
          style="height: 550px;"
        >
          <div
            class="messages w-100 pl-4 pr-4"
          >
            <div />
          </div>
        </div>
        <div
          class="row bg-light"
          style="height: 55px;"
        >
          <form
            class="w-100 pl-2 pr-2"
          >
            <div
              class="input-group mt-2"
            >
              <input
                class="form-control"
                disabled=""
                name="text"
                required=""
                type="text"
                value="another task"
              />
              <div
                class="input-group-append"
              >
                <button
                  class="btn btn-info btn-sm message"
                  disabled=""
                  type="submit"
                >
                  Send
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
